src/query-builder/join-builder.ts

```diff
@@ -79,6 +79,37 @@ export class JoinBuilder<DB, TB extends keyof DB>
     })
   }
 
+  /**
+   * Adds a `using` clause to the query.
+   *
+   * This clause is a non-standard shorthand for the specific situation where both
+   * sides of the join use the same name for the joining column(s).
+   *
+   * ### Examples:
+   *
+   * ```ts
+   * await db
+   *   .selectFrom('person')
+   *   .innerJoin('pet', join => join.using(['name']))
+   *   .selectAll()
+   *   .executeTakeFirstOrThrow()
+   * ```
+   *
+   * The generated SQL (PostgreSQL):
+   *
+   * ```sql
+   * select *
+   * from "person"
+   * inner join "pet" using ("name")
+   * ```
+   */
+  using(columns: [C, ...C[]]): JoinBuilder<DB, TB, C> {
```

Comment:
Does it make sense to have an overload for a single column? I noticed that other APIs like `select` have this.
```js
.innerJoin('pet', join => join.using('col'))
.innerJoin('pet', join => join.using(['col1', 'col2', 'col3']))
```

Reply:
Seems OK to me.
I'm not sure about `[C, ...C[]]`, I don't think we're THAT strict in other places.